src/dbnode/persist/fs/msgpack/decoder.go

```diff
@@ -350,12 +350,19 @@ func (dec *Decoder) decodeIndexBloomFilterInfo() schema.IndexBloomFilterInfo {
 
 func (dec *Decoder) decodeIndexEntry(bytesPool pool.BytesPool) schema.IndexEntry {
 	var opts checkNumFieldsOptions
-	if dec.legacy.decodeLegacyV1IndexEntry {
+	switch dec.legacy.decodeLegacyIndexEntryVersion {
+	case legacyEncodingIndexEntryVersionV1:
 		// V1 had 5 fields.
 		opts.override = true
 		opts.numExpectedMinFields = 5
 		opts.numExpectedCurrFields = 5
+	case legacyEncodingIndexEntryVersionV2:
+		// V2 had 6 fields.
+		opts.override = true
+		opts.numExpectedMinFields = 5
+		opts.numExpectedCurrFields = 6
```

Comment:
nit: may want to error out on default

Reply:
(and I guess add a `case legacyEncodingIndexEntryVersionV3: break` noop)

Reply:
I may be misunderstanding the flow here if that's not a valid case though

Reply:
Will update this flow to make more clear